telemetry/statsd/src/main/scala/io/buoyant/telemetry/statsd/StatsDStatsReceiver.scala

```diff
@@ -0,0 +1,49 @@
+package io.buoyant.telemetry.statsd
+
+import java.util.concurrent.ConcurrentHashMap
+import com.timgroup.statsd.StatsDClient
+import com.twitter.finagle.stats.{Counter, Stat, StatsReceiverWithCumulativeGauges}
+import scala.collection.JavaConverters._
+
+private[telemetry] object StatsDStatsReceiver {
+  // from https://github.com/researchgate/diamond-linkerd-collector/
+  private[statsd] def mkName(name: Seq[String]): String = {
```

Comment:
it looks like `:` are replaced by `_` below.  Not sure if this covers all invalid characters.  Maybe a whitelist is better than a blacklist?